From: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Subject: maple_tree: remove repeated sanity checks from mas_wr_append()
Date: Tue, 18 Jun 2024 13:47:49 -0700

These sanity checks are now redundant as they are already checked in
mas_wr_store_type().  We can remove them from mas_wr_append().

Link: https://lkml.kernel.org/r/20240618204750.79512-16-sidhartha.kumar@oracle.com
Signed-off-by: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Reviewed-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/maple_tree.c |   12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

--- a/lib/maple_tree.c~maple_tree-remove-repeated-sanity-checks-from-mas_wr_append
+++ a/lib/maple_tree.c
@@ -4061,17 +4061,9 @@ static inline unsigned char mas_wr_new_e
 static inline bool mas_wr_append(struct ma_wr_state *wr_mas,
 		unsigned char new_end)
 {
-	struct ma_state *mas;
+	struct ma_state *mas = wr_mas->mas;
 	void __rcu **slots;
-	unsigned char end;
-
-	mas = wr_mas->mas;
-	if (mt_in_rcu(mas->tree))
-		return false;
-
-	end = mas->end;
-	if (mas->offset != end)
-		return false;
+	unsigned char end = mas->end;
 
 	if (new_end < mt_pivots[wr_mas->type]) {
 		wr_mas->pivots[new_end] = wr_mas->pivots[end];
_
